packages/middleware-sdk-ec2/src/index.ts

```diff
@@ -123,6 +125,9 @@ export function copySnapshotPresignedUrlMiddleware(options: PreviouslyResolved):
           if (!(request.body ?? "").includes("PresignedUrl=")) {
             request.body += `&PresignedUrl=${extendedEncodeURIComponent(args.input.PresignedUrl)}`;
           }
+          if (typeof options.bodyLengthChecker === "function") {
+            request.headers["content-length"] = String(options.bodyLengthChecker(request.body));
+          }
```

Comment:
This isn't strictly necessary, since the content length middleware was handling this when running after this middleware. 

However, explicitly setting the content length in the same code path where the request body is being directly modified is safer and gives more information to the reader.